lambdalisue/vim-gita
# RFC: Fix noshellslash issue on Windows (Need help of Windows user!!!)

To fix #40. While I don't have Windows, need your help to fix this issue!
#### How to do unit tests

**It is not requried but better.**

First of all, vim-gita use [themis.vim](https://github.com/thinca/vim-themis) to test all its features.
So please follow the [instruction](https://github.com/thinca/vim-themis/blob/master/doc/themis.txt#L57) of themis.vim to test vim-gita. (Never mind 13 tests in `Pending`, these tests cannot be performed as unit test).
#### How to do behavior tests

vim-gita have a lot of features which cannot be tested via unit test.
the following commands required to be manually tests (everything should be performed under a root directory of a git repository of vim-gita)
At least, `:Gita file HEAD autoload\gita.vim` and `:Gita browse` is required.
**Please check the following when you confirm the command success**, otherwise please paste the exception. You may like [Capture.vim](https://github.com/tyru/capture.vim) which capture the output of Ex command and create a new buffer so you can easily copy & paste.
- [x] `:Gita add` on untracked files
- [x] **`:Gita browse` on `autoload\gita.vim`**
- [x] `:Gita checkout HEAD autoload\gita.vim`
- [x] `:Gita commit` after you stage changes
- [x] `:Gita commit` to actual commit
- [x] `:Gita diff HEAD~50` on `autoload\gita.vim`
- [x] `:Gita diff HEAD~50 --window=double` on `autoload\gita.vim`
- [x] `:Gita diff-ls origin/HEAD`
- [x] **`:Gita file HEAD autoload\gita.vim`**
- [x] `:Gita reset HEAD -- autoload\gita.vim`
- [x] `:Gita rm autoload\gita.vim`
- [x] `:Gita status` after you edit something

Thanks for your great cooperations !
